### PR TITLE
chore: Drop scala java8 compat

### PR DESCRIPTION
**Pull request by sebastian-alfers**
Fixes https://github.com/akka/akka-http/issues/4417

Imports seem inconsistent, so its a best-effort. Should maybe auto-alignt in a separate pr?

Is there maybe a better way to avoid the "map to object", like this one?
```
.map(length -> (Object)length));
```

### CODE REVIEW

**Comment by sebastian-alfers** [comment]
Currently fails with:
```
[info] - should Work for usage when running HTTPS server *** FAILED *** (42 milliseconds)
[info]   The future returned an exception of type: javax.net.ssl.SSLHandshakeException, with message: NotAfter: Tue Sep 03 13:18:30 CEST 2024. (SSLContextFactorySpec.scala:105)
```

But seems also to fail in `main`, so taking a look what is going on there.

**Comment by sebastian-alfers** [comment]
Depends on https://github.com/akka/akka-http/pull/4419.

File: akka-http-core/src/main/java/akka/http/javadsl/model/ContentRange.java [hunk 1]
```diff
@@ -30,7 +31,7 @@ public static ContentRange create(long first, long last, long instanceLength) {
     }
     @SuppressWarnings("unchecked")
     public static ContentRange create(long first, long last, OptionalLong instanceLength) {
-        return ContentRange$.MODULE$.apply(first, last, OptionConverters.toScala(instanceLength));
+        return ContentRange$.MODULE$.apply(first, last, OptionConverters.toScala(instanceLength).map(length -> (Object)length));
```
Comment (johanandren) [inline akka-http-core/src/main/java/akka/http/javadsl/model/ContentRange.java:34]: `OptionConverters.toScala(OptionalLong)` returns `Option[java.lang.Long]` so this is just Java not understanding variance (as expected), however, it is surprising that it thinks that `Option[Long]` in Scala is `Option<Object>`.

Actually mapping involves allocation and a lambda that will actually be invoked, we don't want that, just force-cast is probably ok `(Option<Object>)(Object) OptionConverters.toScala(instanceLength)` even if it feels a bit dirty.

File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/CacheDirectives.java [hunk 2]
```diff
@@ -19,7 +19,7 @@ public static CacheDirective MAX_STALE() {
         return new akka.http.scaladsl.model.headers.CacheDirectives.max$minusstale(OptionConverters.toScala(Optional.empty()));
     }
     public static CacheDirective MAX_STALE(long deltaSeconds) {
-        return new akka.http.scaladsl.model.headers.CacheDirectives.max$minusstale(OptionConverters.toScala(OptionalLong.of(deltaSeconds)));
+        return new akka.http.scaladsl.model.headers.CacheDirectives.max$minusstale(OptionConverters.toScala(OptionalLong.of(deltaSeconds)).map(delta -> (Object)delta));
```
Comment (johanandren) [inline akka-http-core/src/main/java/akka/http/javadsl/model/headers/CacheDirectives.java:22]: This seems kinda unnecessary, can't we just call `new Some<>(deltaSeconds)` and skip the whole conversion?

File: akka-http-core/src/main/java/akka/http/javadsl/model/ContentRange.java (hunk 1 above)
Comment (johanandren) [reply akka-http-core/src/main/java/akka/http/javadsl/model/ContentRange.java:34]: Or if we want to keep it slightly cleaner, this should compile fine `(instanceLength.isPresent() ? Option.apply(instanceLength.getAsLong()) : Option.empty())`

File: akka-http-core/src/main/java/akka/http/javadsl/model/sse/ServerSentEvent.java [hunk 3]
```diff
@@ -18,18 +18,19 @@
...
  * Representation of a server-sent event. According to the specification, an empty data field
  * designates an event which is to be ignored which is useful for heartbeats.
  */
 public abstract class ServerSentEvent {
 
-    private static final Option<String> stringNone =  toScala(Optional.empty());
+    private static final Option<String> stringNone = OptionConverters.toScala(Optional.empty());
 
-    private static final Option<Object> intNone = toScala(OptionalInt.empty());
+    private static final Option<Object> intNone = OptionConverters.toScala(OptionalInt.empty()).map(emptyIntOption -> (Object)emptyIntOption);
```
Comment (johanandren) [inline akka-http-core/src/main/java/akka/http/javadsl/model/sse/ServerSentEvent.java:33]: These two are really weird to start with, `Option.empty()` should be fine for both?

**Review by johanandren — APPROVED** [review]

**Pull request merged**